operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -1029,7 +1029,15 @@ private boolean isNonClusteredServer(String serverName) {
       }
 
       private boolean isRunning(@Nonnull String serverName) {
-        return RUNNING_STATE.equals(getRunningState(serverName));
+        return RUNNING_STATE.equals(getRunningState(serverName)) && isRollCompleteFor(serverName);
+      }
+
+      private boolean isRollCompleteFor(@Nonnull String serverName) {
+        return isAdminServer(serverName) || rollStateComputation.isRollCompleteFor(serverName);
```

Comment:
Why we don't need to calculate the rolling state for admin server?

Reply:
We don't roll the admin server. 

Rolling is something we do with the managed servers to prevent too many from being down at the same time. That means that it is very possible that we know that some managed servers still need to be replaced, but we're not going to do that yet, until others have been. But there is only one admin server, so we replace it immediately when we know we need to.

Reply:
Really?  If I changed the restartVersion,  I saw all servers rolled including the admin server.  Am I missing something about the use case? @doxiao?

Reply:
I expect that you saw all of the server pods _restarted_. We using _rolling_ to mean that the restarts are spread out over time. If the admin server restart doesn't happen for a minute or so after some managed servers are restarted, that would be a problem.

Reply:
@jshum2479 , Yes, when restartVersion is changed for the domain, all servers will be restarted one by one, and the domain can only be marked as Completed when all servers reach the expected state. I think the confusion is about terminology. I noticed that Russ mentioned that "we replace it (admin server) immediately when we know we need to". In implementation, the restart happened in two different ways: for the admin server, the pod is replaced right away (as Russ mentioned in his response), and for the managed servers, the restart is deferred until, all managed server pods are looked at, and then restarted in a rolling fashion.

Reply:
The question is if we need to verify if the admin server has reached the expected state here? I think we probably should.

Reply:
Under what conditions would the admin server be running at this point, but _not_ yet in the expected state? Remember that we don't even get to the status update until we've created the replacement admin server, if it needs to be replaced.

Reply:
I was thinking about the periodical status update code path as well.

Reply:
And in that code path, the admin server is either not running, or is up-to-date (pending a future make-right).